packages/core/src/browser/common-frontend-contribution.ts

```diff
@@ -114,6 +114,11 @@ export namespace CommonCommands {
         label: 'Save All'
     };
 
+    export const QUIT: Command = {
```

Comment:
Yes actually the plan would be to have it quit the inner-most command, then maybe the app ? vscode seems to work like that but I find that a bit error prone.

For the keybindings it doesn't need any implementation since basically the way it cancels a keysequence now is that it makes sure that there can be no key sequence that has Ctrl-Q in it. Thus when you type Ctrl-Q you type an invalid keysequence and it quits.